src/view/com/composer/select-language/SelectPostLanguagesDialog.tsx

```diff
@@ -0,0 +1,418 @@
+      const uncheckedRecent = recentLanguages.filter(lang => !isChecked(lang))
+      const all = allowedLanguages
+        .filter(lang => !recentLanguagesCode2.includes(lang.code2))
+        .filter(lang => !isInRecents(lang))
+
+      return {
+        all,
+        checkedRecent,
+        uncheckedRecent,
+      }
+    }
+  }, [
+    allowedLanguages,
+    search,
+    langPrefs.postLanguageHistory,
+    checkedLanguagesCode2,
+  ])
+
+  const listHeader = (
+    <View
+      style={[
+        {
+          paddingBottom: 8,
+          backgroundColor: t.atoms.bg.backgroundColor,
+        },
+        isNative && a.pt_2xl,
+      ]}>
+      <Text
+        nativeID="dialog-title"
+        style={[t.atoms.text, a.text_left, a.font_bold, a.text_xl, a.mb_sm]}>
+        <Trans>Choose Post Languages</Trans>
+      </Text>
+      <Text
+        nativeID="dialog-description"
+        style={[t.atoms.text_contrast_medium, a.text_left, a.text_md, a.mb_lg]}>
+        <Trans>Select up to 3 languages used in this post</Trans>
+      </Text>
+      <View style={[a.w_full, a.flex_row, a.align_stretch, a.gap_xs, a.pb_0]}>
+        <View style={[a.flex_1, a.relative]}>
+          <TextField.Root>
+            <TextField.Input
+              value={search}
+              onChangeText={setSearch}
+              placeholder="Search languages"
+              style={[a.pl_xl]}
+              maxLength={50}
+              label="Search languages"
+            />
+            <View
+              style={[
+                a.absolute,
+                a.top_0,
+                a.bottom_0,
+                a.justify_center,
+                a.pl_0,
+                a.z_10,
+                isNative && a.pl_sm,
+                {pointerEvents: 'none'},
+              ]}>
+              <MagnifyingGlassIcon
+                strokeWidth={3}
+                size={16}
+                color={
+                  search.length > 0
+                    ? t.palette.primary_500
+                    : t.atoms.text_contrast_low.color
+                }
+                style={{
+                  marginLeft: 'auto',
+                  marginRight: 'auto',
+                }}
+              />
+            </View>
+          </TextField.Root>
+        </View>
+        {showSearchCancel && (
+          <Button
+            label="Cancel"
+            size="small"
+            variant="ghost"
+            color="secondary"
+            style={[a.px_sm]}
+            onPress={() => setSearch('')}>
+            <ButtonText>
+              <Trans>Cancel</Trans>
+            </ButtonText>
+          </Button>
+        )}
+      </View>
+    </View>
+  )
+
+  const isCheckedRecentEmpty =
+    displayedLanguages.checkedRecent.length > 0 ||
+    displayedLanguages.uncheckedRecent.length > 0
+
+  const isDisplayedLanguagesEmpty = displayedLanguages.all.length === 0
+
+  const flatListData = [
+    ...(isCheckedRecentEmpty ? [{type: 'header', label: 'Recently used'}] : []),
```

Comment:
Remember translations!

```suggestion
    ...(isCheckedRecentEmpty ? [{type: 'header', label: _(msg`Recently used`)}] : []),
```